src/genai/extensions/langchain/utils.py

```diff
@@ -111,3 +112,21 @@ def __add__(self, other: GenerationChunk) -> CustomGenerationChunk:
             )
         else:
             raise TypeError(f"unsupported operand type(s) for +: '{type(self)}' and '{type(other)}'")
+
+
+class CustomAIMessageChunk(AIMessageChunk):
+    """Replaces LangChain's 'merge_dicts' with our simplified 'merge_objects' utility"""
```

Comment:
this was a bug? (just curious)

Reply:
It depends. Their 'merge_dicts' function throws when dictionaries contain an attribute with the `DateTime` data type (in our case, `created_at`)—it does not replace the attribute with the newer one as we do.